libbeat/outputs/elasticsearch/client.go

```diff
@@ -178,6 +177,25 @@ func eventBulkMeta(index string, event common.MapStr) bulkMeta {
 	return meta
 }
 
+// getIndex returns the full index name
+// Index is either defined in the config as part of the output
+// or can be overload by the event through setting index
+func getIndex(event common.MapStr, index string) string {
+
+	ts := time.Time(event["@timestamp"].(common.Time)).UTC()
+
+	// Check if index is ste dynamically
+	if dynamicIndex, ok := event["index"]; ok {
```

Comment:
Yes, I'd like it as meta-information. Then we don't need the type assertion and we'll probably do the same for `@timestamp`.